client/src/main/scala/io/delta/sharing/client/DeltaSharingClient.scala

```diff
@@ -871,14 +871,10 @@ class DeltaSharingRestClient(
           0L
         }
       },
-      respondedFormat = respondedFormat,
-      includeEndStreamActionHeader = includeEndStreamActionHeader,
+      respondedFormat = getRespondedFormat(capabilities),
```

Comment:
Optional optimization:
Can getResponse itself create the capabilitesMap to avoid duplicate parsing and map creation from capabilities, since it seems the capabilities map is what is used everywhere?

Reply:
updated.